packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -56,14 +57,34 @@ export class AwsS3Publish implements PublisherBase {
     // or AWS shared credentials file at ~/.aws/credentials will be used to authenticate
     // https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-environment.html
     // https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-shared.html
-    const credentials = config.getOptionalConfig(
+    const credentialsConfig = config.getOptionalConfig(
       'techdocs.publisher.awsS3.credentials',
     );
     let accessKeyId = undefined;
     let secretAccessKey = undefined;
-    if (credentials) {
-      accessKeyId = credentials.getOptionalString('accessKeyId');
-      secretAccessKey = credentials.getOptionalString('secretAccessKey');
+    let credentials: Credentials | CredentialsOptions | undefined = undefined;
+    if (credentialsConfig) {
+      const roleArn = credentialsConfig.getOptionalString('roleArn');
+      if (roleArn && aws.config.credentials instanceof Credentials) {
+        credentials = new aws.ChainableTemporaryCredentials({
+          masterCredentials: aws.config.credentials as Credentials,
+          params: {
+            RoleSessionName: 'backstage-aws-organization-processor',
```

Comment:
Maybe it should be `backstage-aws-techdocs-s3-publisher` or something like that here? 
Great to see this support being added, especially for the AWS organization processor.

Reply:
Oh yes! Good catch - this should be something techdocs related.